storm-core/test/jvm/org/apache/storm/stats/TestStatsUtil.java

```diff
@@ -92,6 +113,33 @@ public void makeTopoInfo() {
         worker2Resources.put(new WorkerSlot("node3", 3), ws3);
     }
 
+    private Map<String, Object> createBeatBoltStats() {
+        return createBeatStats("bolt");
+    }
+
+    private Map<String, Object> createBeatSpoutStats() {
+        return createBeatStats("spout");
+    }
+
+    private Map<String, Object> createBeatStats(final String type) {
+        Map<String, Object> stats = new HashMap<>();
+        stats.put("type", type);
+
+        stats.put("acked", new HashMap<>());
```

Comment:
Bolt and spout have different stats
https://github.com/apache/storm/blob/master/storm-server/src/main/java/org/apache/storm/stats/StatsUtil.java#L2049-L2085

It would be nice to reflect that here even though the information is not really used  .

Reply:
Updated to only include the fields required for each component type.